examples/IOC_data.ipynb

```diff
@@ -35,13 +35,6 @@
    "cell_type": "markdown",
    "id": "f3d14b77",
    "metadata": {
-    "execution": {
-     "iopub.execute_input": "2022-06-08T12:43:45.973799Z",
-     "iopub.status.busy": "2022-06-08T12:43:45.973432Z",
-     "iopub.status.idle": "2022-06-08T12:43:51.596147Z",
-     "shell.execute_reply": "2022-06-08T12:43:51.595523Z",
-     "shell.execute_reply.started": "2022-06-08T12:43:45.973779Z"
-    },
```

Comment:
I agree that it's nice to have small file footprint in the repo, but wouldn't it be more helpful to the users to see the actual outputs without running the notebooks? Isn't there a better way than stripping all the outputs?

Reply:
Disclaimer: I really like jupyter as a REPL to play with the code and prototype but I think it is really bad for pretty much anything else.

IMHV there is no really good solution for this. 

If you keep the output of the cells, you run the risks of having outdated output. To avoid that you need to use a different pre-commit hook which will re-execute the notebooks each time you do a git commit. This will keep the notebooks in sync, but you run into a different issue. Each time you execute the notebooks, the timestamps of the cell do change. Therefore you will have to `git add` the *.ipynb files to each and every commit. Either that , or you will have to find a way to figure out if the only thing that changed was the timestamps or not. I don't know of such a tool, and even if I knew, this feels like a huge [XY problem](https://meta.stackexchange.com/questions/66377/what-is-the-xy-problem) to me. 

And there is another issue. Git repo size. Since most of the time notebooks contain images `git adding` new/updated notebooks results in commits in the size of MBs (for big enough notebooks that is). You can very easily explode your git repo size. In one of the repos we had with @brey due to notebooks full of images we ended up having commits up to 50MBs (re-execute all cells, git add, git commit = bingo!)

People have tried to solve this in various ways. For example [jupytext](https://github.com/mwouts/jupytext) and [nbdime](https://nbdime.readthedocs.io/en/latest/) are two projects I am aware of. I am sure there are more. These might or might not work fine. I deliberately try not to use notebooks for anything else but prototyping, so I don't have much incentive to properly test them. In all honesty the whole problem feels like trying to put a square peg in a round hole. You say, "I must use notebooks", and then you try find a way to make notebooks work for you.

A somewhat better solution might be to try to convert the notebooks to HTML and save that instead. This can also be easily automated with [nbconvert](https://nbconvert.readthedocs.io/en/latest/) or something similar. and since what you get is plain HTML diffs etc should mostly work. You can even set a pre-commit hook, so that your HTML pages are kept in sync. But if you go down this road, why don't you put your examples in sphinx in the first place?

Anyway, Sorry for the long rant. We don't make such a big usage of notebooks, so I guess we can just omit nbstripout

Reply:
@pmav99 thanks for the detailed explanation. It makes sense. I'm also concerned about the size of notebooks with output. I think putting some sample outputs in the documentation as well as adding a Binder link will help with the user learning concern that I had.